### PR TITLE
1630 extreme event@devel

### PR DESCRIPTION
**Pull request by SyedMubasheer**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_param_extreme_event.R [hunk 1]
```diff
@@ -321,28 +321,53 @@ derive_param_extreme_event <- function(dataset = NULL,
...
-  # Create output dataset
-  bind_rows(dataset, new_obs)
+    new_obs <- bind_rows(events, noevents) %>%
+      mutate(
+        !!!set_values_to
+      )
+
+    # Create output dataset
+    bind_rows(dataset, new_obs)
+  }
```
Comment (bundfussr) [inline R/derive_param_extreme_event.R:372]: Please avoid duplication of code. Only the creation of `new_var` should be conditional. I.e.
```
  events <- dataset_source %>%
    filter_if(filter_source) %>%
    filter_extreme(
      by_vars = subject_keys,
      order = order,
      mode = mode,
      check_type = check_type
    )
  
  noevents <- anti_join(
    select(dataset_adsl, intersect(source_vars, adsl_vars)),
    select(events,!!!subject_keys),
    by = sapply(subject_keys, as_name)
  )

  if (!is.null(new_var)) {
    events <- mutate(events,!!new_var := true_value)
    noevents <- mutate(noevents,!!new_var := false_value)
  }
```

File: tests/testthat/test-derive_param_extreme_event.R [hunk 2]
```diff
@@ -168,3 +169,43 @@ test_that("derive_param_extreme_event Test 3: latest evaluable tumor assessment
     keys = c("USUBJID", "PARAMCD", "ADT")
   )
 })
+
+## Test 4:
+## derive latest evaluable tumor assessment date parameter without overwriting existing result ----
+test_that("derive_param_extreme_event Test 4: latest evaluable tumor assessment date parameter", {
```
Comment (bundfussr) [inline tests/testthat/test-derive_param_extreme_event.R:175]: Please format according to the unit test guidance (https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#set-up-the-test-script). You can achieve this easily by using the ["Format test_that test file" addin](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#addin-admiraldevformat_test_that_file).

**Review by bundfussr — CHANGES_REQUESTED** [review]
Please add an item to the changelog (`NEWS.md`).

File: tests/testthat/test-derive_param_extreme_event.R (hunk 2 above)
Comment (SyedMubasheer) [reply tests/testthat/test-derive_param_extreme_event.R]: Thanks for the review @bundfussr!! I will update and re request for a review!

**Review by bundfussr — APPROVED** [review]

**Comment by SyedMubasheer** [comment]
Thank you for the approval @bundfussr  :)

**Comment by bms63** [comment]
Thank you Syed!

**Comment by SyedMubasheer** [comment]
> Thank you Syed!

Anytime Ben :)

**Pull request merged**